bus/api.go

```diff
@@ -198,3 +198,14 @@ type UploadParams struct {
 	TotalShards   uint8
 	ContractSet   string
 }
+
+type RedundancySettings struct {
+	MinShards   int
+	TotalShards int
+}
+
+func DefaultRedundancySettings() (rs RedundancySettings) {
```

Comment:
I'll leave it up to @lukechampine to make the decision but my vote would be on keeping default values in `main.go` for when we pass the command line flags.

Reply:
I felt having these settings as command line flags is awkward because they're updatable through the API, so that would be confusing in my opinion. Also if I manually updated the setting through the API, and I would rerun the command or whatever to run the `renterd` binary, I'd have to remember to update the command line flag.

That said, I don't mind updating it at all if you both disagree with me.

Reply:
hmm. We definitely want to be able to change these settings at runtime via the API. But it should also be possible to set them via flags or a config file. (Imagine deploying a fresh `renterd` instance; should you have to wait for it to finish loading and then call API endpoints in order to configure it?)

Surely we can't be the first project to grapple with this problem. I wonder what others do. 🤔

Reply:
Since you both feel all settings should be configurable through CLI options I think there's no other option then to add that at least, so I did. I do still feel that the UX could be better because I can easily imagine a scenario where someone updates redundancy settings through the API, and forgets to update his ansible script or whatever and ends up (silently) overwriting his settings.

We should probably add a `logger` to the bus and print a big `WARNING` when we see we are overwriting settings with default values. Or maybe we should simply not overwrite settings with default values... (although that's kind of weird behaviour as well 🤷‍♂️ )

Reply:
@lukechampine so apparently people use [`SIGHUP`](https://en.wikipedia.org/wiki/Signal_(IPC)#SIGHUP) for this, which was news to be but apparently it's a relatively well known pattern. 

> Many [daemons](https://en.wikipedia.org/wiki/Daemon_(software)) (who have no controlling terminal) interpret receipt of this signal as a request to reload their configuration files and flush/reopen their logfiles instead of exiting.

I'm not sure if it's something worth implementing at this point. I think having the settings in the database has many benefits over a configuration file. I was surprised to learn though that people are using this signal for hot reloading configuration files.

Few examples of projects that do this hostapd, sshd, snort, ...